### PR TITLE
DDF-3006 Fix bug introduced in camel-amqp that prevents it from starting

### PR DESCRIPTION
**Pull request by harrison-tarr**
#### What does this PR do?
Fix bug introduced in camel-amqp that prevents it from starting
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@ryeats 
@emmberk 
@vinamartin 
@roelens8 
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Core APIs](https://github.com/orgs/codice/teams/core-apis)
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jaymcnallie
@kcwire
#### How should this be tested? (List steps with links to updated documentation)
ensure that you can `feature:install camel-amqp` _and_ `feature:install artemis-amqp`. The order matters.
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-3006
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: platform/platform-app/src/main/resources/features.xml [hunk 1]
```diff
@@ -94,16 +95,25 @@
         <feature version="${camel.version}">camel-core</feature>
         <bundle>mvn:org.apache.camel/camel-spring/${camel.version}</bundle>
     </feature>
+
+    <!--
+        The following two bulletted changes were made to allow camel-amqp to start without resolution conflicts
```
Comment (harrison-tarr) [inline platform/platform-app/src/main/resources/features.xml]: spelling on `bulletted`? not sure if it should be `bulletted` or `bulleted`

File: platform/platform-app/src/main/resources/features.xml [hunk 2]
```diff
@@ -45,6 +45,7 @@
         - added camel as feature prerequisite in camel-http to keep things ordered
         - added cxf as feature prerequisite in camel to keep things ordered
         - removed features that are not needed
+        - updated versions of qpid and proton versions and added netty bundles in camel-amqp
```
Comment (emmberk) [inline platform/platform-app/src/main/resources/features.xml]: Also add a bullet for the change in the `camel-jms` feature

File: catalog/ui/catalog-ui-search/yarn.lock [hunk 3]
```diff
@@ -688,7 +688,7 @@ arraytools@^1.0.0, arraytools@^1.1.2:
   version "1.1.2"
   resolved "https://registry.yarnpkg.com/arraytools/-/arraytools-1.1.2.tgz#410f8e8137b67475e53c1918572e78d893f11fbf"
 
-arrify@^1.0.0:
+arrify@^1.0.0, arrify@^1.0.1:
```
Comment (ryeats) [inline catalog/ui/catalog-ui-search/yarn.lock]: I don't think this should be updating, did you intend to modify this file

Comment (harrison-tarr) [inline catalog/ui/catalog-ui-search/yarn.lock]: i asked @andrewkfiedler about this, he said I should commit it.

File: platform/platform-app/src/main/resources/features.xml [hunk 4]
```diff
@@ -167,9 +177,11 @@
         <feature version="${camel.version}">camel-core</feature>
         <feature version="${spring-version-range}">spring</feature>
         <feature version="${spring-version-range}">spring-jms</feature>
-        <bundle dependency="true">mvn:org.apache.geronimo.specs/geronimo-jta_1.1_spec/${geronimo-jta-spec-version}</bundle>
+        <!-- Use version 1.2 of javax transaction, DDF update to the camel feature file -->
+        <feature>transaction</feature>
+        <!-- Use version 2.0 of javax jms, DDF update to the camel feature file -->
+        <bundle dependency="true">mvn:org.apache.geronimo.specs/geronimo-jms_2.0_spec/${geronimo-jms_2_spec.version}</bundle>
```
Comment (emmberk) [inline platform/platform-app/src/main/resources/features.xml]: This property should be geronimo-jms2-spec-version instead (https://github.com/codice/ddf/blob/master/platform/platform-app/pom.xml#L346).

Comment (emmberk) [inline platform/platform-app/src/main/resources/features.xml]: We're trying to keep everything named the same as the original file that these features were copied from: `apache-camel-2.19.0-src/parent/pom.xml`.

Comment (emmberk) [inline platform/platform-app/src/main/resources/features.xml]: And move `<geronimo-jms_2_spec.version>1.0-alpha-2</geronimo-jms_2_spec.version>` property in the root pom to this section of the root pom: https://github.com/codice/ddf/blob/ac97624deeb4e454247631a5c1211df51e6d37ab/pom.xml#L121-L129.

File: platform/platform-app/pom.xml [hunk 5]
```diff
@@ -348,11 +350,11 @@
         <jackson2-version>2.8.8</jackson2-version>
```
Comment (emmberk) [inline platform/platform-app/pom.xml]: `geronimo-jms-spec-version` property is no longer being used in the platform-app camel features. It should be removed from this list

Comment (harrison-tarr) [inline platform/platform-app/pom.xml]: It is used via property replacement in the cxf features though. Should I just replace the `${geronimo-jms-spec-version}` at https://github.com/codice/ddf/blob/master/platform/platform-app/pom.xml#L387 with `1.1.1` (what geronimo-jms-spec-version is set to)?

Comment (emmberk) [inline platform/platform-app/pom.xml]: 👍 Good point. I missed that. Yep, it should probably stay then.

File: platform/platform-app/src/main/resources/features.xml (hunk 4 above)
Comment (harrison-tarr) [inline platform/platform-app/src/main/resources/features.xml]: I'm not sure that `geronimo-jms_2_spec.version` should be moved to the `cxf` section of dependencies, it's an `artemis `dependency (used in the broker app and in the camel-sjms platform feature, it looks like)

Comment (emmberk) [inline platform/platform-app/src/main/resources/features.xml]: This property should be `geronimo-jms2-spec-version` instead.

We're trying to keep everything named the same as the original file that these features were copied from https://github.com/apache/camel/blob/camel-2.19.0/pom.xml.

And move `<geronimo-jms_2_spec.version>1.0-alpha-2</geronimo-jms_2_spec.version>` property in the root pom to [this section of the root pom](https://github.com/codice/ddf/blob/ac97624deeb4e454247631a5c1211df51e6d37ab/pom.xml#L121-L129).

And modify the `<geronimo-jms2-spec-version>1.0-alpha-2</geronimo-jms2-spec-version>` in the platform-app pom to be `<geronimo-jms2-spec-version>${geronimo-jms_2_spec.version}</geronimo-jms2-spec-version>`, like the other properties in [that section of the root pom](https://github.com/codice/ddf/blob/ac97624deeb4e454247631a5c1211df51e6d37ab/pom.xml#L121-L1290).

File: platform/platform-app/src/main/resources/features.xml (hunk 2 above)
Comment (harrison-tarr) [inline platform/platform-app/src/main/resources/features.xml]: Fixed
and added a block comment on the `camel-jms` feature

File: platform/platform-app/src/main/resources/features.xml (hunk 4 above)
Comment (emmberk) [inline platform/platform-app/src/main/resources/features.xml]: The intent of that section in the root pom is to force some versions that are used elsewhere in DDF to be the same versions that are started in the Camel and CXF features. Is this version not one of those cases?

Comment (harrison-tarr) [inline platform/platform-app/src/main/resources/features.xml]: fixed

File: platform/platform-app/src/main/resources/features.xml (hunk 1 above)
Comment (shaundmorris) [inline platform/platform-app/src/main/resources/features.xml]: Thought it was `bulleted`

Comment (harrison-tarr) [inline platform/platform-app/src/main/resources/features.xml]: Me too, but Intellij didn't like it so I switched it to `bulletted` but Intellij didn't like that either. But okay, that's one vote for `bulleted` and none for `bulletted` so I'll switch it back! 👍

Comment (harrison-tarr) [inline platform/platform-app/src/main/resources/features.xml]: Fixed

**Comment by shaundmorris** [comment]
build now

**Pull request merged**